src/github.com/stellar/gateway/listener/payment_listener.go

```diff
@@ -166,40 +176,37 @@ func (pl *PaymentListener) onPayment(payment horizon.PaymentResponse) (err error
 		OperationID: payment.ID,
 		ProcessedAt: pl.now(),
 		PagingToken: payment.PagingToken,
+		Status:      "Processing...",
 	}
 
-	return pl.process(payment, dbPayment)
-}
-
-func (pl *PaymentListener) process(payment horizon.PaymentResponse, dbPayment *entities.ReceivedPayment) (err error) {
-	savePayment := func(payment *entities.ReceivedPayment) (err error) {
-		pl.log.Info(payment.Status)
-		err = pl.entityManager.Persist(payment)
+	err = pl.entityManager.Persist(dbPayment)
+	if err != nil {
 		return
 	}
 
+	dbPayment.Status, err = pl.process(payment)
+
+	pl.log.WithFields(logrus.Fields{"status": dbPayment.Status, "err": err}).Error("Payment processed")
```

Comment:
We shouldn't be using the `Error` severity here...  there is a possibility that the status returned from process is `"Success"`